### ISSUE TITLE
MQTT command and file transfer

### ISSUE THREAD

**Issue by philcali** [philcali/device-pool#53]
## Background

Add an MQTT data plane integration. It looks like with [MQTT file based delivery][1] we can propose an optional file transfer method.

- [x] Create an MQTT client (make adaptable, but default to aws provided)
- [x] Create a Shadow document connection
- ~~Create a MQTT connections / content transfer factory~~

[1]: https://docs.aws.amazon.com/iot/latest/developerguide/mqtt-based-file-delivery.html

**Comment by philcali** [comment]
MQTT is the obvious way to send and react to commands, but there is another way that doesn't involve involve long lived connections: Shadow API.

Control can be done through the HTTP calls client side. The original option would have to support a special "control" certificate. Really, both options could be supported.

**Comment by philcali** [comment]
Reading deeper about the MQTT transport: it's intended to send files to a mass of IoT devices, ie: over the air update of microcontrollers. The underlying backbone of S3 is still the same. Therefore, there is no advantage of supporting the MQTT protocol for file transfers. I guess it would be possible to implement a file stream server, but once again, provides no clear benefit. To that end: I've decided to remove the MQTT protocol file transfer out into a separate issue, and unblock the 1.2.0 release.
